spring-cloud-zuul-ratelimit-core/src/main/java/com/marcosbarbero/cloud/autoconfigure/zuul/ratelimit/config/properties/RateLimitProperties.java

```diff
@@ -16,35 +16,62 @@
 
 package com.marcosbarbero.cloud.autoconfigure.zuul.ratelimit.config.properties;
 
-import static com.marcosbarbero.cloud.autoconfigure.zuul.ratelimit.config.properties.RateLimitProperties.PREFIX;
-
-import java.util.LinkedHashMap;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import java.util.List;
 import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import javax.validation.constraints.NotNull;
 import lombok.Data;
+import lombok.NoArgsConstructor;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.validation.annotation.Validated;
 
 /**
  * @author Marcos Barbero
  */
 @Data
-@ConfigurationProperties(PREFIX)
+@Validated
+@NoArgsConstructor
+@ConfigurationProperties(RateLimitProperties.PREFIX)
 public class RateLimitProperties {
 
     public static final String PREFIX = "zuul.ratelimit";
 
-    private Map<String, Policy> policies = new LinkedHashMap<>();
-
+    private Policy defaultPolicy;
+    @NotNull
+    private Map<String, Policy> policies = Maps.newHashMap();
```

Comment:
Why not to use `Collections.emptyMap()` instead?

Reply:
Spring properties is inserting to the map, not creating new one